### ISSUE TITLE
Bug report: simulaton error regarding LIDAR

### ISSUE THREAD

**Issue by foxton9** [OpenFAST/openfast#1606]
**Bug description**
When LIDAR is enabled, the simulation will fail and throw an error. I have attempted various configurations, but the same error persists.

**To Reproduce**
1. configure the LIDAR parameters in `InflowWind` dat file
```
================== LIDAR Parameters ===========================================================================
          3   SensorType          - Switch for lidar configuration (0 = None, 1 = Single Point Beam(s), 2 = Continuous, 3 = Pulsed)
          2   NumPulseGate        - Number of lidar measurement gates (used when SensorType = 3)
         30   PulseSpacing        - Distance between range gates (m) (used when SensorType = 3)
          1   NumBeam             - Number of lidar measurement beams (0-5)(used when SensorType = 1)
       -200   FocalDistanceX      - Focal distance co-ordinates of the lidar beam in the x direction (relative to hub height) (only first coordinate used for SensorType 2 and 3) (m)
          0   FocalDistanceY      - Focal distance co-ordinates of the lidar beam in the y direction (relative to hub height) (only first coordinate used for SensorType 2 and 3) (m)
          0   FocalDistanceZ      - Focal distance co-ordinates of the lidar beam in the z direction (relative to hub height) (only first coordinate used for SensorType 2 and 3) (m)
0.0 0.0 0.0   RotorApexOffsetPos  - Offset of the lidar from hub height (m)
         18   URefLid             - Reference average wind speed for the lidar[m/s]
       0.25   MeasurementInterval - Time between each measurement [s]
      False   LidRadialVel        - TRUE => return radial component, FALSE => return 'x' direction estimate
          1   ConsiderHubMotion   - Flag whether to consider the hub motion's impact on Lidar measurements
```
2. start simulation with OpenFast v3.5.0
3. observe the following error messages:

```
FAST_InitializeAll:SrvD_Init:BladedInterface_Init:EXavrSWAP_Init:Error allocating memory for the
u%LidSpeed array; array was already allocated.
EXavrSWAP_Init:Error allocating memory for the u%LidSpeed array; array was already allocated.

 OpenFAST encountered an error during module initialization.
 Simulation error level: FATAL ERROR
```

**Expected behavior**
The simulation can be executed successfully.

**Screenshots**
![image](https://github.com/OpenFAST/openfast/assets/135503227/38c7f13d-58f1-4324-bae6-841f83dd2794)

**OpenFAST Version**
A screenshot of the OpenFAST system description is attached above.
The OpenFast binary files are downloaded from [OpenFast v3.5.0](https://github.com/OpenFAST/openfast/releases/tag/v3.5.0).

**System Information:**
 - OS: Windows 10
 - Compiler: Intel® Fortran Compiler 2022.1.0
 - Compiler settings: default settings

**Additional context**
The error message suggests a potential mistake in allocating during the initialization of the `ServoDyn` module.
I inspected the source code and debugged it using Visual Studio, which has given me some insights into the possible cause. 

I'm uncertain if this is the actual root cause, as it could be attributed to my incorrect  OpenFast input configurations.

The `u%LidSpeed` is allocated in `ServoDyn.f90`:
https://github.com/OpenFAST/openfast/blob/6b8706b75698cb1c97c6a8e3e9f42538a63fa71b/modules/servodyn/src/ServoDyn.f90#L389

and the it is allocated in `BladedInterface_EX.f90` again:

https://github.com/OpenFAST/openfast/blob/6b8706b75698cb1c97c6a8e3e9f42538a63fa71b/modules/servodyn/src/BladedInterface_EX.f90#L380

In addition, the code for allocating `u%MsrPositionsX`, `u%MsrPositionsY` and `u%MsrPositionsZ` in `BladedInterface_EX.f90`  leads to the same problem.

One possible solution might be
```fortran
if (allocated(InitInp%LidSpeed)) then
   if ( .not. allocated(u%LidSpeed)) then
      CALL AllocAry(u%LidSpeed, size(InitInp%LidSpeed), 'u%LidSpeed', errStat2, ErrMsg2)
      if (Failed())  return
   endif
   u%LidSpeed = InitInp%LidSpeed
endif
```

After implementing the aforementioned change, the simulation is able to complete and the obtained output appear reasonable, but the speed is noticeably slower compared to the case when LIDAR is disabled. Is this considered aberrant?

**Comment by andrew-platt** [comment]
@foxton9, thanks for reporting this!  This is indeed a bug that was introduced right before the release.  This will be corrected in the bugfix release 3.5.1 that we intend to post this Friday.

**Comment by andrew-platt** [comment]
This should be fixed with #1620.

As for the speed issue, I don't really know how much of a speed penalty to expect.

**Comment by andrew-platt** [comment]
Merged into rc-3.5.1 (release date slipped to later this month -- we have a few additional things to include).

Thanks @foxton9 for posting a solution: I used it almost directly.
